Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -253,7 +269,14 @@ void FunctionDefinitionWriter::visit(AST::TypeDecl& type)
 void FunctionDefinitionWriter::visit(AST::ArrayType& type)
 {
     ASSERT(type.maybeElementType());
-    ASSERT(type.maybeElementCount());
+
+    if (!type.maybeElementCount()) {
+        m_stringBuilder.append("device ");
+        visit(*type.maybeElementType());
+        m_stringBuilder.append("*");
+        return;
+    }
```

Comment:
Isn't this wrong, as per our discussion this morning?

Reply:
Yes, I left a comment above, I fixed this in a separate PR and will remove from this one before merging.